src/format/OpData01.cpp

```diff
@@ -0,0 +1,143 @@
+#include "OpData01.h"
+
+#include "crypto/CryptoHash.h"
+#include "crypto/SymmetricCipher.h"
+
+#include <QMessageLogger>
+#include <QtGui/QtGui> // for qWarning() et al
+
+#define RETURN_OPDATA01_ERROR_INSTANCE(msg)                                                                            \
+    {                                                                                                                  \
+        auto result = new OpData01();                                                                                  \
+        result->m_error = true;                                                                                        \
+        result->m_errorStr = new QString((msg));                                                                       \
+        return result;                                                                                                 \
+    }
+
+OpData01* OpData01::decode01Base64(QString const& b64String, const QByteArray& key, const QByteArray& hmacKey)
+{
+    const QByteArray b64Bytes = QByteArray::fromBase64(b64String.toUtf8());
+    return decode01(b64Bytes, key, hmacKey);
+}
+
+OpData01* OpData01::decode01(const QByteArray& data, const QByteArray& key, const QByteArray& hmacKey)
+{
+    // qDebug("OpData01.decode-ing %d bytes using a %d sized key, %d sized HMAC\n", data.size(), key.size(),
+    // hmacKey.size());
+
+    const QByteArray header("opdata01");
+
+    QByteArray::const_iterator it = data.cbegin();
+    QByteArray::const_iterator it_end = data.cend();
+    if (!data.startsWith(header)) {
+        RETURN_OPDATA01_ERROR_INSTANCE(QString("data claims it does not start with header: %1")
+                                           .arg(QString::fromUtf8(data.mid(0, header.size()).toHex())));
+    } else {
+        it += header.size();
+    }
+
+    qlonglong qlen(0LL);
+    int len;
+    {
```

Comment:
What is this scope for? I don't see any variable declarations directly inside of it.